Source/WebKit/WPEPlatform/wpe/WPEGestureController.cpp

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (C) 2024 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WPEGestureController.h"
+
+#include <cmath>
+#include <optional>
+#include <wtf/glib/WTFGType.h>
+
+namespace WPE {
+
+class GestureDetector final {
+public:
+    WPEGesture detectGesture(WPEEvent* event)
+    {
+        WPEGesture detectedGesture = WPE_GESTURE_NONE;
+
+        switch (wpe_event_get_event_type(event)) {
+        case WPE_EVENT_TOUCH_DOWN:
+            if (double x, y; wpe_event_get_position(event, &x, &y))
+                m_touchDownPoint = Point { x, y };
```

Comment:
I would just call this `m_origin` or something like that, touch is redundant I think. We should probably save the sequence id, to check in future events that they belong to the current sequence.

Reply:
I also think we can assume this is a TAP, and change it later if it ends up being a drag. Since we handle the gestures on UP, it's should be ok.